protocol/shannon/http_client.go

```diff
@@ -0,0 +1,297 @@
+package shannon
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptrace"
+	"net/url"
+	"runtime"
+	"sync/atomic"
+	"time"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+)
+
+// httpClientWithTracing provides HTTP client functionality with built-in request tracing,
+// metrics collection, and detailed logging for debugging timeout and connection issues.
+type httpClientWithTracing struct {
```

Comment:
Why not `httpClientWithDebugMetrics`? I feel like individual request tracing is the one thing this client doesn't do.